File: FrontEnd/src/services/AdminService.js
import axios from 'axios';

const BASE_PATH_URL="http://localhost:8080/admin";

class AdminService{

    adminLogin(email,password){
        return axios.post(BASE_PATH_URL+`/login/${email}/${password}`)
    }
    

    getAllUsers(){
    return axios.get(BASE_PATH_URL+"/getAllUsers")
    }

    deleteUser(id){
        return axios.delete(BASE_PATH_URL+`/user/delete/${id}`)
    }
    
    addUser(name,email,password,role){
        return axios.post(BASE_PATH_URL+"/user/register",{name,email,password,role}) 
    }

    updateUser(userId,name,email,password,role){
        return axios.put(BASE_PATH_URL+`/user/update/${userId}`,{name,email,password,role})
    }

   getAllPatients(){
    return axios.get(BASE_PATH_URL+"/getAllPatients");
   }

   addPatient(name,age,gender,contact,medicalhistory){
    return axios.post(BASE_PATH_URL+"/addPatient",{name,age,gender,contact,medicalhistory})
   }

   updatePatient(patientId,name,age,gender,contact,medicalhistory){
    return axios.put(BASE_PATH_URL+`/Patient/update/${patientId}`,{name,age,gender,contact,medicalhistory})
   }
   
   deletePatient(id){
    return axios.delete(BASE_PATH_URL+`/Patient/delete/${id}`)
   }


   getAllReports(){
    return axios.get(BASE_PATH_URL+"/getAllReports");
   }

   addReport(userId,patientId,testResults,diagnosisInfo,date){
    return axios.post(BASE_PATH_URL+"/addReport",{userId,patientId,testResults,diagnosisInfo,date})
   }

   updateReport(reportId,userId,patientId,testResults,diagnosisInfo,date){
    return axios.put(BASE_PATH_URL+`/Report/update/${reportId}`,{userId,patientId,testResults,diagnosisInfo,date})
   }

   deleteReport(id){
    return axios.delete(BASE_PATH_URL+`/Report/delete/${id}`)
   }

   getAllAppointments(){
    return axios.get(BASE_PATH_URL+"/getAllAppointments")
   }

   addAppointment(uid,pid,date){
    return axios.post(BASE_PATH_URL+"/request",{uid,pid,date})
   }

}

export default AdminService

File: BackEnd/src/main/java/com/manipal/demo/service/AdminService.java
package com.manipal.demo.service;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import com.manipal.demo.dto.AptDto;
import com.manipal.demo.dto.ReportDto;
import com.manipal.demo.entity.Admin;
import com.manipal.demo.entity.Appointment;
import com.manipal.demo.entity.Patient;
import com.manipal.demo.entity.Report;
import com.manipal.demo.entity.User;

@Service
public interface AdminService {
	public Admin adminLogin(String email, String password);
	   public String adminLogout(Long id);
	
	   public User userRegister(User user);
	   public Optional<Patient> findPatientById(long id);
	  // public Optional<Patient> findPatientByName(String name);
	   public List<Patient> getAllPatients();
	   public List<User> getAllUsers();
	   public void deleteUserById(long id);
	   public void updateUserById(long id,User user);
	 
	   public void addPatient(Patient patient);
	   public void deletePatientById(long id);
	   public void updatePatient(Long id,Patient patient);
	   
	   public List<Report> getAllReports();
	   public String addReport(ReportDto reportDto);
	   public List<Report> findReportByUserName(String name);
	   public void deleteReportById(long id);
	   public void updateReport(Long r_id,ReportDto reportDto);
	   
	   public String addAppointment(AptDto aptdto);
	   public void updateAppointment(Long apt_id,AptDto aptdto);
	   public List<Appointment> getAllAppointments();
}
